Dev

iredb opened the pull request:
Задания выполнены. Рабочая версия проекта.

Stradalec committed:
very bad version of work with cli, login and not secured password (Files are necessary. If training manual is not exist - all is legally)

Stradalec committed:
Add some arguments + replace files

Stradalec committed:
Add some try to use salt

Stradalec committed:
add checks to password and login, list of users (hash replaced to wrong)

Stradalec committed:
some small fixes

Stradalec committed:
fix: small bug with not done part

iredb committed:
добавлен класс ресурса

реализация класса ресурса, тестовые объекты в main

Stradalec committed:
fix: fixed not working permisson

Stradalec committed:
feature: add necessary exit processes

Stradalec committed:
feature: add compile.sh

Stradalec committed:
feature: add start.sh

Stradalec committed:
test: add test.sh

Stradalec committed:
docs: update Readme.md

Stradalec committed:
chore: delete jar.file

Winnie-the-Pooh2019 commented:
- [ ] Логика аутентификации, настройка мок-данных, выдача разрешений, все это происходит в мейне.
- [ ] Экзит-коды лучше в энамах хранить, так ничего не понятно.
- [ ] `Resource` и хранит данные, и управляет разрешениями
- [ ] в классе `Resource` же и хранятся данные

Stradalec committed:
#1, point 2: exit codes now in enum

Stradalec committed:
#1 point 1: add function to create mock data and authorization

Stradalec committed:
#1, point 3. Permission moved to other file

iredb committed:
анализ и предложения по рефакторингу проекта

Добавлены makrdown файлы, содержащие краткий анализ актуальной структуры проекта на соответствие принципам SOLID и подходу чистой архитектуры. Также предлагается план по рефакторингу.

Stradalec committed:
docs: add (steal from official developers) documentation about set up Sonar

Stradalec committed:
docs: add some details to Sonar integration document

iredb committed:
Update CleanArchitecture.md

iredb committed:
в этом фрагменте кода ничего не произошло.

Winnie-the-Pooh2019 committed:
Merge pull request #2 from Stradalec/dev-refactor

Dev refactor

Stradalec committed:
chore: add sonar properties file

Stradalec committed:
Merge branch 'dev-refactor' of https://github.com/Stradalec/HLPT-Lab2- into dev-refactor

Stradalec committed:
Merge pull request #4 from Stradalec/dev-refactor

Add sonar file (now in correct branch)

Stradalec committed:
chore: fixing crimes

Stradalec committed:
tests: add tests

Stradalec committed:
chore: clean some files

Stradalec committed:
refactor: try to make pass Sonar test check

Stradalec committed:
tests are useless

Stradalec committed:
refactor: authorization and hash now in  class AuthService

iredb committed:
IAuthService, IPermissionManager

Stradalec committed:
fix: add overrides to classes

Stradalec committed:
refactor: add CommandHandler class

Stradalec committed:
chore: fix some bad style

Stradalec committed:
feature: add IResourceRepository

Stradalec committed:
chore:  task completed

iredb committed:
попытка в ресурс

iredb committed:
IResource, class Resource, class App.

дореализована реализация ресурса. основной (единственный) сценарий работы приложения вынесен в отдельный класс App.

Stradalec committed:
fix: minor fix

Winnie-the-Pooh2019 commented:
Что все в одном файле...

Stradalec committed:
refactor: mega file is gone, now we have some little files

iredb committed:
изгнаны ненужные комментарии

Stradalec committed:
refactor: now files more organise and crimes against humanity no longer exist

Stradalec committed:
chore: docs now live in other place

Stradalec committed:
chore: scripts move to directory and fixed documentation

Stradalec committed:
chore: reorganise files

Stradalec committed:
docs: add tests plan (first version)

Stradalec committed:
tests: add necessary files to tests, Resource tests, necessary libs and docs

Stradalec committed:
tests: add tests to permission manager

Stradalec committed:
tests: add tests to AuthService

Stradalec committed:
docs: add some data

Stradalec committed:
fix: fix

Stradalec committed:
docs: update README.md

iredb committed:
добавлен commandHandlerTest.kt

Stradalec committed:
fix: small changes

Winnie-the-Pooh2019 commented:
Ну во-первых, ридмик должен быть в корне проекта

Winnie-the-Pooh2019 commented:
На данном этапе класс `CommandHandler` делает слишком много дел:
- и парсит аргументы
- и аутентификацию проводит
- и валидацию данных и тд и тп
- и вообще и швец и жнец и на дуде игрец
- и вообще это все один метод
Короче, где-то плачет SRP


К тому же:
- сервисы в пакете с моделями, это зачем?
- еще и джарник засунули в пакет с тест-классами...

Stradalec committed:
refactor: add new structure

Stradalec committed:
refactor: removed non testable code,

iredb committed:
CommandHandler -> parser, useCase

iredb committed:
изгнан exitStrategy

Stradalec committed:
fix: small fixes to new files

Stradalec committed:
feature: now some userRepository and other cool things. Update Create MockData. Working tests in next commit

Stradalec committed:
fix: tests now alive

Stradalec committed:
chore: add library and update compile script

Stradalec committed:
refactor: some update to UserData

Stradalec committed:
docs: first version of plan and start of refactor

Stradalec committed:
fix: fix permission

Stradalec committed:
refactor: userData now User

Stradalec committed:
refactor: minor changes: resource repo, permission repo, some kind of refactor, head explosion, new mockData. Need to clean this
